### DISCUSSION TITLE
Programmatically start HTTP server from command line application

### DISCUSSION THREAD

**Discussion by libe** [quarkusio/quarkus#43925] [Q&A] [answered]
Hello,

I am new to Quarkus and I am writing a command line application that has a command "serve" the starts up the HTTP server and waits for requests and shut down. The other commands allows to connect to the server and call its endpoints.  Does anybody can provide some hints on how to do this? I would like to use the Picocli integration so that I have other nice CLI features enabled. 

Thanks in advance!

**Comment by cescoffier** [comment, marked as answer]
You can create your own HTTP server using the Vert.x API directly.

Here is an example:

1. Add the quarkus-vertx extension
2. Write something like:

```java
@Inject Vertx vertx;
// ...
var server = vertx.createHttpServer();
server
      .requestHandler(req -> req.response().end("Hello verticle 2"))
      .listen(8080);
```

**Reply by libe** [reply]
Thanks for the tip it works!
